sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3854,6 +3854,38 @@ object functions {
   @scala.annotation.varargs
   def map_concat(cols: Column*): Column = withExpr { MapConcat(cols.map(_.expr)) }
 
+  /**
+   * Parses a column containing a CSV string into a `StructType` with the specified schema.
+   * Returns `null`, in the case of an unparseable string.
+   *
+   * @param e a string column containing CSV data.
+   * @param schema the schema to use when parsing the CSV string
+   * @param options options to control how the CSV is parsed. accepts the same options and the
+   *                CSV data source.
+   *
+   * @group collection_funcs
+   * @since 3.0.0
+   */
+  def from_csv(e: Column, schema: StructType, options: Map[String, String]): Column = withExpr {
+    CsvToStructs(schema, options, e.expr)
+  }
+
+  /**
+   * (Java-specific) Parses a column containing a CSV string into a `StructType`
+   * with the specified schema. Returns `null`, in the case of an unparseable string.
+   *
+   * @param e a string column containing CSV data.
+   * @param schema the schema to use when parsing the CSV string
+   * @param options options to control how the CSV is parsed. accepts the same options and the
+   *                CSV data source.
+   *
+   * @group collection_funcs
+   * @since 3.0.0
+   */
+  def from_csv(e: Column, schema: String, options: java.util.Map[String, String]): Column = {
```

Comment:
@MaxGekk, can we replace `schema: String` to `schema: Column` for `schema_of_csv`?

Reply:
you can create a new type characterOrstructOrColumn? (letter casing is weird)

but I wasn't following - what's the case for schema is a Column?

Reply:
> but I wasn't following - what's the case for schema is a Column?

The `schema_of_json`/`schema_of_csv` functions take a string in a column and return inferred schema of JSON/CSV as a column (in DDL format). To combine `from_json`/`from_csv` with `schema_of_json`/`schema_of_csv` need to pass a schema as the `Column` type. See in Scala : https://github.com/apache/spark/blob/1007cae20e8f566e7d7c25f0f81c9b84f352b6d5/sql/core/src/test/scala/org/apache/spark/sql/JsonFunctionsSuite.scala#L396

Reply:
Eh, I wasn't following too. Is the problem related to its parameters for the function? we can just define `characterOrColumn` and use it. We can do something like:

```
if # is character
  schema <- lit(schema)
else if # is column
  schema <- schema
else
  stop("it should be column or characters")
```

like you did in Python side.